File: LeetCode_Solutions/LeetCode_Solutions/NextGreaterElementI_496.m
//
//  NextGreaterElementI_496.m
//  LeetCode_Solutions
//
//  Created by Huni on 29/03/2017.
//  Copyright © 2017 KnightJoker. All rights reserved.
//


#import "NextGreaterElementI_496.h"

@implementation NextGreaterElementI_496

- (NSMutableArray *)nextGreaterElement:(NSArray *)findNums
                                  nums:(NSArray *)nums {

    NSMutableArray *returnArray = [[NSMutableArray alloc] init];
    for (NSNumber *temp in findNums) {
        for (NSNumber *num in nums) {
            if (temp < num && temp == num) {
                [returnArray addObject:@(-1)];
            } else {
                [returnArray addObject:num];
            }
        }
    }
    return returnArray;
}

@end

File: LeetCode_Swift/LeetCode_Swift/NextGreaterElementI_496.swift
//
//  NextGreaterElementI_496.swift
//  LeetCode_Swift
//
//  Created by Huni on 25/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class NextGreaterElementI_496 {

    func nextGreaterElement(_ findNums: [Int], _ nums: [Int]) -> [Int] {
        
        var returnArray = Array<Int>()
        var dic = Dictionary<Int, Int>()
        
        for findNum in findNums {
            for num in nums {
                if findNum == num {
                    var tempIndex = nums.index(of: num)
                    while tempIndex! < nums.count {
                        if nums[tempIndex!] > num {

                            dic[findNums.index(of: num)!] = nums[tempIndex!]
                            break
                        }
                        tempIndex = tempIndex! + 1
                    }
                }
            }

        }
        
        var i = 0
        while i < findNums.count {
            returnArray.append(-1)
            i += 1
        }
        
        for (a,b) in dic {
            returnArray[a] = b
        }

        return returnArray
    }
}

